#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# nullspace_impedance_param
gen.add("task1_Kp", double_t, 0, "task1_Kp", 50, 0, 400)#
gen.add("task1_Kv", double_t, 0, "task1_Kv", 20, 0, 30)#

gen.add("task2_Md", double_t, 0, "inertia", 3, 0, 100)#
gen.add("task2_Bd", double_t, 0, "damping", 8, 0, 100)
gen.add("task2_Kd", double_t, 0, "stiffness", 4, 0, 100)

# nullspace_impedance_EBobserver_param
gen.add("K_pos", double_t, 0, "K_pos", 80, 0, 400)#
gen.add("K_ori", double_t, 0, "K_ori", 8, 0, 200)#
gen.add("P_pos", double_t, 0, "P_pos", 2, 0, 200)#
gen.add("P_ori", double_t, 0, "P_ori", 1, 0, 100)#
gen.add("gamma", double_t, 0, "gamma", 0.1, 0, 400)#

# nullspace_impedance_MBobserver_param
gen.add("Kp_pos", double_t, 0, "Kp_pos", 80, 0, 400)#
gen.add("Kp_ori", double_t, 0, "Kp_ori", 8, 0, 20)#
gen.add("Kv_pos", double_t, 0, "Kv_pos", 2, 0, 200)#
gen.add("Kv_ori", double_t, 0, "Kv_ori", 1, 0, 10)#

#normal
gen.add("Md", double_t, 0, "inertia", 3, 0, 100)#
gen.add("Bd", double_t, 0, "damping", 8, 0, 100)
gen.add("Kd", double_t, 0, "stiffness", 4, 0, 100)


#pkgname nodename name
exit(gen.generate(PACKAGE, "dynamic_compliance", "nullspace_impedance_param"))
